husky_directory/blueprints/search.py

```diff
@@ -35,111 +39,104 @@ class RenderingContext(DirectoryBaseModel):
 class SearchBlueprint(Blueprint):
     @inject
     def __init__(self, logger: Logger, injector: Injector):
-        super().__init__("search", __name__, url_prefix="/search")
+        super().__init__("search", __name__)
         self.injector = injector
         self.logger = logger
-        self.add_url_rule("/", view_func=self.get, methods=("GET",))
-        self.add_url_rule("/", view_func=self.render, methods=("POST",))
-        # When no output format is provided, defaults to html
+        self.add_url_rule("/", view_func=self.index, methods=("GET",))
+        self.add_url_rule("/", view_func=self.search_listing, methods=("POST",))
         self.add_url_rule(
-            "/person/<href_token>",
-            defaults={"output_format": "html"},
-            view_func=self.get_person,
-            methods=("GET", "POST"),
+            "/person/listing", view_func=self.get_person_listing, methods=("POST",)
         )
-        # /person/{token}/vcard
-        # defaults to 'html' (above)
         self.add_url_rule(
-            "/person/<href_token>/<output_format>",
-            view_func=self.get_person,
-            methods=("GET", "POST"),
+            "/person/vcard",
+            view_func=self.get_person_vcard,
+            methods=("POST",),
```

Comment:
This, along with the changes to `.html` files, represents the juiciest part of this effort, which was to redefine the routes so that functionality is separated, and the right render is being used for the right query.

The only thing it changes for the _user_ is that the `/search` suffix will no longer appear in their search history, an improvement, since before a GET at that endpoint would just redirect to `/` anyway.